internal/internal_workflow.go

```diff
@@ -1239,7 +1239,7 @@ func (w *WorkflowOptions) getSignalChannel(ctx Context, signalName string) Recei
 	if ch, ok := w.signalChannels[signalName]; ok {
 		return ch
 	}
-	ch := NewBufferedChannel(ctx, defaultSignalChannelSize)
+	ch := NewNamedBufferedChannel(ctx, signalName, defaultSignalChannelSize)
```

Comment:
I hope there are no side effects for this change. `name` is used for error message and it would be great to see something meaningful there rather then empty string.